src/api/lib/station.js

```diff
@@ -1,21 +1,47 @@
 import { readStations } from 'mav-stations';
 
+/**
+ * get a station matching by name
+ *
+ * @param name of station
+ * @returns station found by name
+ */
 const station = async (name) => {
   // eslint-disable-next-line prefer-promise-reject-errors
   if (!name) return Promise.reject(false);
 
   return new Promise((resolve, reject) => {
-    const stations = readStations;
-    stations().on('data', (station) => {
-      if (station.name === name || station.aliasNames.indexOf(name) !== -1) {
-        resolve(station);
-        stations().removeAllListeners();
-      }
-    });
-    stations().on('end', () => {
-      reject('MÁV station not found');
-    });
+    const stations = readStations();
+    stations
+      .on('data', (station) => {
+        if (station.name === name || station.aliasNames.indexOf(name) !== -1) {
+          resolve(station);
+          stations().removeAllListeners();
```

Comment:
```suggestion
          stations.removeAllListeners();
```
Vermutlich so, da du den Funktionsaufruf in Zeile 14 vorgezogen hast, oder?

Reply:
Gefixt.